javascripts/discourse/api-initializers/discourse-mermaid-theme-component.js

```diff
@@ -58,9 +59,55 @@ async function applyMermaid(element, key = "composer") {
       mermaid.dataset.processed = true;
       mermaid.querySelector(".spinner")?.remove();
     }
+
+    if (key === "composer") {
+      debounce(() => updateMarkdownHeight(mermaid, index), 500);
+    }
   });
 }
 
+function updateMarkdownHeight(mermaid, index) {
+  let height = parseInt(mermaid.getBoundingClientRect().height);
+  let calculatedHeight = parseInt(mermaid.dataset.calculatedHeight);
+
+  if (height === 0) {
+    return;
+  }
+
+  if (height !== calculatedHeight) {
+    mermaid.dataset.calculatedHeight = height;
+    // TODO: need to use API here
+    let composer = document.getElementsByClassName("d-editor-input")[0];
+    let old = composer.value;
```

Comment:
Not sure why `old` is needed just for the split? Did you mean to use it later down in `(joined !== composer.value)`? If so `oldComoserValue` would be more descriptive, if not IMO just remove it and use `composer.value`

Reply:
true, yeah will remove